### ISSUE TITLE
pedal function

### ISSUE THREAD

**Issue by felixroos** [tidalcycles/strudel#292]
https://strudel.tidalcycles.org/?mFjklRqDqgNg

maybe also rename to hold..

**Comment by felixroos** [comment]
also good for strumming: https://strudel.tidalcycles.org/?yXqOXLVIc79c

**Comment by felixroos** [comment]
todo fix:

`sequence(pure(1),silence).pedal(pure(1)).queryArc(0.5,1)` > `Array []` 
"you could fix it by querying the pedals with the whole of the hap
to be sure of finding everything
being sure to throw any events away that don't intersect with the original query"

**Comment by yaxu** [comment]
It'd be good to find three related names for the different behaviours:
* extend offset (current 'pedal')
* extend onset
* extend both

Ideas for the first two:
* fore/aft
* head/tail

For the last one I was thinking maybe expand/grow, but the original haps could also be made smaller.. so maybe
* fill
* shape

Or as it's related to `struct`
* structFill
* ...

Or we could also name the three variants together
* fillOn
* fillOff
* fill

These are really new alignment functions, so we can add them to the operators, making `add.fill` etc or whatever we call them.

**Comment by felixroos** [comment]
I like `head` / `tail`, maybe also `fillRight` / `fillLeft` which is more explicit

**Comment by yaxu** [comment]
I guess `head` implies losing the tail. `fillRight` / `fillLeft` is good although we've moved away from implying that time goes from left to right, which is quite nice.. 

Maybe just `onset`, `offset` and `fill` works?

Or more radically, `on` and `off`, renaming current `off` to `offset`?

**Comment by felixroos** [comment]
> we've moved away from implying that time goes from left to right, which is quite nice..

makes sense, maybe `headFill`, `tailFill` ? Or `prolong` and `prelong` ?

> Maybe just onset, offset and fill works?

To me, onset and offset sound too simple compared to what this is doing.

**Comment by yaxu** [comment]
Although actually I'm not sure how this _should_ work. My guess:
* `"a ~".pedal("1")` -> `"a"`
* `"a ~".pedal("1")` -> `"a"`
* `"a".pedal("1 ~")` -> `"a ~"` ?
* `"a@3 ~".pedal("~ 1")` -> `"a".mask("~ 1")` - e.g. a fragment with its first half missing?

**Comment by yaxu** [comment]
> To me, onset and offset sound too simple compared to what this is doing.

I was thinking that `.offset()` applies a pattern of offsets?

**Comment by felixroos** [comment]
maybe it helps (or futher complicates) to include the versions of the function that use the other sides of the timespans:

- quantize offsets to end  `"a ~ ~ ~".x("1 1")` -> `a ~` aka `tailFill` (pedal)
- quantize offsets to start `"~ a@2 ~".x("1 1")` -> `~ a ~ ~` aka `tailCut`
- quantize onsets to end `"~ a@2 ~".x("1 1")` -> `~ ~ a ~` aka `headCut`
- quantize onsets to start `"~ a ~ ~".x("1 1")` -> `a ~` aka `headFill`

**Comment by yaxu** [comment]
Heh, the right metaphor must be out there somewhere..
